controllers/tf_controller_plan.go

```diff
@@ -59,6 +59,12 @@ func (r *TerraformReconciler) plan(ctx context.Context, terraform infrav1.Terraf
 		planRequest.Destroy = true
 	}
 
+	// check if Lock Timeout is set
+	if terraform.Spec.TFState.LockTimeout.Duration.String() != "" {
```

Comment:
We got nil reference here in the test: 
https://github.com/weaveworks/tf-controller/actions/runs/3854570045/jobs/6699132334

Could you please add an if to check that TFState is not nil here?

The frame:
```
E0114 13:17:24.579832   18289 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 608 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x2b654c0?, 0x423b820})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.25.4/pkg/util/runtime/runtime.go:75 +0x99
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:110 +0xb9
panic({0x2b654c0, 0x423b820})
	/opt/hostedtoolcache/go/1.19.4/x64/src/runtime/panic.go:884 +0x212
github.com/weaveworks/tf-controller/controllers.(*TerraformReconciler).plan(_, {_, _}, {{{0x29439c9, 0x9}, {0xc000a46880, 0x20}}, {{0xc0010c4210, 0x26}, {0x0, ...}, ...}, ...}, ...)
	/home/runner/work/tf-controller/tf-controller/controllers/tf_controller_plan.go:63 +0x417
github.com/weaveworks/tf-controller/controllers.(*TerraformReconciler).reconcile(_, {_, _}, {_, _}, {{{0x29439c9, 0x9}, {0xc000a46880, 0x20}}, {{0xc0010c4210, ...}, ...}, ...}, ...)
```

Reply:
thank you for flagging this @chanwit , will my last push fix it?
apologies for the the lack of attention on this PR too